meta-iotqa/lib/oeqa/runtime/multimedia/vaapi/test_vaapi_encode_decode.py

```diff
@@ -0,0 +1,198 @@
+import os
+from oeqa.oetest import oeRuntimeTest
+
+'''
+This test suit tests VAAPI Decoding and Encoding
+'''
+
+
+class VAAPITest(oeRuntimeTest):
+
+    '''
+    The video files have been created using gstreamer and videotestsrc as input
```

Comment:
You can simplify the code quite a bit if you do the following:
- on the target check for available encoders and decoders
- for each format: check both encoder and decoder exists, encode a sample file and then decode it
- if the check is false, skip the test for that platform